cxplat/src/cxplat_winuser/memory_winuser.cpp

```diff
@@ -165,31 +170,36 @@ __drv_allocatesMem(Mem) _Must_inspect_result_ _Ret_writes_maybenull_(new_size) v
 }
 
 void
-cxplat_free(_Frees_ptr_opt_ void* pointer)
+cxplat_free_any_tag(_Frees_ptr_opt_ void* pointer)
 {
-    if (pointer == nullptr)
-    {
+    if (pointer == nullptr) {
         return;
     }
     if (_cxplat_leak_detector_ptr) {
         _cxplat_leak_detector_ptr->unregister_allocation(reinterpret_cast<uintptr_t>(pointer));
     }
     cxplat_allocation_header_t* header = _header_from_pointer(pointer);
-    if (header->pool_type == CxPlatNonPagedPoolNxCacheAligned)
-    {
+    if (header->pool_type == CxPlatNonPagedPoolNxCacheAligned) {
         uint8_t* memory_block = _memory_block_from_aligned_pointer(pointer);
         _aligned_free(memory_block);
-    }
-    else
-    {
+    } else {
         uint8_t* memory_block = _memory_block_from_unaligned_pointer(pointer);
         free(memory_block);
     }
 }
 
+void
+cxplat_free(_Frees_ptr_opt_ void* pointer, uint32_t tag)
```

Comment:
Not something to fix in this PR, but we went through the decision process when we made out abstraction layer on whether or not to accept NULL pointers, and it was ultimately decided not to, and push that requirement to the app, so as to eliminate the requirement of that branch. Many time the app has a cleanup function that only runs if the pointer is non-null, then it does a lot of cleanup work before actually freeing. Then this checks again. I know your answer will likely be if/when we inline this the compiler should deal with that, but you also know my feelings on depending on the compiler. 😄

Reply:
When we went through the decision process, we came to the opposite conclusion (ALL free apis should accept nullptr), for mainly the following reasons:
1) Consistency with the POSIX standard.  This is the industry consensus on good API design, which is why free() is defined to accept a null ptr.   Thus, programmers familiar with standard API conventions aren't surprised.
2) Minimizes number of code paths (reducing code complexity and increasing code coverage), since otherwise many places, instead of 1, have to test for null.

If oyu disagree, feel free to follow up with the standards bodies who have had consensus for decades :)